### PR TITLE
add custom spark image spec

### PR DESCRIPTION
**Pull request by elmiko**

### CODE REVIEW

File: custom-spark-images.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+is deployed by the Oshinko suite of applications. For example, the
+oshinko-rest server can accept an environment variable which defines the
+Spark image it should use for deployment. But, this is only on a per-project,
+per-installation basis.
+
+Additionally, even when the image is changed during Oshinko installation,
+this image is fixed for all subsequent deployments of Spark in that project.
+
+These configurations restrain a user from being able to modify the Spark
+image on a per-application basis. Further, they require reinstallation of
```
Comment (tmckayus) [inline custom-spark-images.md]: I might say "redeployment" instead of reinstallation here because you can swap out the image with just an oc edit

**Comment by tmckayus** [comment]
looks pretty good to me. What do you think about adding logging to oshinko-rest so that when a cluster is created with a non-default spark image, we log that explicitly?  Name of the cluster, pull spec for the image? That might make diagnosing compatibility errors easier to diagnose ...

**Comment by elmiko** [comment]
which logging are you referring to?

if it's the debug option on oshinko-rest, i don't have an objection to turing that on but it will mainly just add the content of the REST calls to the log output.

**Comment by elmiko** [comment]
re: logging

adding the spark image and pull spec seems like a really nice idea either way

**Comment by tmckayus** [comment]
I think under standard INFO logging, I would note the pull spec for the default image when oshinko starts up, and then note the pull spec for any cluster created with a non-default spark image.

But that doesn't necessarily have to be part of this spec

**Comment by elmiko** [comment]
that sounds reasonable to me, i can add some language to that effect or we can just agree to put it in the implementation.

**Comment by tmckayus** [comment]
agreeing to put it in the impl is fine

**Pull request merged**